specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2019-09-30-preview/openShiftManagedClusters.json

```diff
@@ -0,0 +1,817 @@
+          "description": "ID of the Vnet created for OSA cluster."
+        }
+      },
+      "description": "Represents the OpenShift networking configuration"
+    },
+    "OSType": {
+      "type": "string",
+      "default": "Linux",
+      "enum": [
+        "Linux",
+        "Windows"
+      ],
+      "x-ms-enum": {
+        "name": "OSType",
+        "modelAsString": true
+      },
+      "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+    },
+    "OpenShiftContainerServiceVMSize": {
+      "type": "string",
+      "x-ms-enum": {
+        "name": "OpenShiftContainerServiceVMSize",
+        "modelAsString": true
+      },
+      "description": "Size of OpenShift VMs.",
+      "enum": [
+        "Standard_D2s_v3",
+        "Standard_D4s_v3",
+        "Standard_D8s_v3",
+        "Standard_D16s_v3",
+        "Standard_D32s_v3",
+        "Standard_D64s_v3",
+        "Standard_DS4_v2",
+        "Standard_DS5_v2",
+        "Standard_F8s_v2",
+        "Standard_F16s_v2",
+        "Standard_F32s_v2",
+        "Standard_F64s_v2",
+        "Standard_F72s_v2",
+        "Standard_F8s",
+        "Standard_F16s",
+        "Standard_E4s_v3",
+        "Standard_E8s_v3",
+        "Standard_E16s_v3",
+        "Standard_E20s_v3",
+        "Standard_E32s_v3",
+        "Standard_E64s_v3",
+        "Standard_GS2",
+        "Standard_GS3",
+        "Standard_GS4",
+        "Standard_GS5",
+        "Standard_DS12_v2",
+        "Standard_DS13_v2",
+        "Standard_DS14_v2",
+        "Standard_DS15_v2",
+        "Standard_L4s",
+        "Standard_L8s",
+        "Standard_L16s",
+        "Standard_L32s"
+      ]
+    },
+    "OpenShiftAgentPoolProfileRole": {
+      "type": "string",
+      "enum": [
+        "compute",
+        "infra"
+      ],
+      "x-ms-enum": {
+        "name": "OpenShiftAgentPoolProfileRole",
+        "modelAsString": true
+      },
+      "description": "OpenShiftAgentPoolProfileRole represents the role of the AgentPoolProfile."
+    },
+    "OpenShiftManagedClusterMasterPoolProfile": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique name of the master pool profile in the context of the subscription and resource group."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of masters (VMs) to host docker containers. The default value is 3."
+        },
+        "vmSize": {
+          "$ref": "#/definitions/OpenShiftContainerServiceVMSize",
+          "description": "Size of agent VMs."
+        },
+        "subnetCidr": {
+          "type": "string",
+          "description": "Subnet CIDR for the peering."
+        },
+        "osType": {
+          "$ref": "#/definitions/OSType",
+          "description": "OsType to be used to specify os type. Choose from Linux and Windows. Default to Linux."
+        }
+      },
+      "required": [
```

Comment:
What's the difference between this Profile and `OpenShiftManagedClusterAgentPoolProfile` which have `name` as required property but here doesn't?

Reply:
The master pool is only one and can easily be given a default name. The Agent pools are 1+ and need names to differentiate them.